__tests__/pet.test.js

```diff
@@ -0,0 +1,135 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+  it('returns an object', () => {
+    expect(new Pet('Fido')).toBeInstanceOf(Object);
+  });
+
+it('pet has a name', () => {
+    const pet = new Pet('Fido');
+    expect(pet.name).toEqual('Fido');
+  });
+
+	it('pet initial age is 0', () => {
+    const pet = new Pet('Fido');
+    expect(pet.age).toEqual(0);
+  });
+
+  it('pet initial hunger is 0', () => {
+    const pet = new Pet('Fido');
+    expect(pet.hunger).toEqual(0);
+  });
+
+  it('pet initial fitness is 10', () => {
+    const pet = new Pet('Fido');
+    expect(pet.fitness).toEqual(10);
+  });
+});
+
+describe('growUp', () => {
+  it('increase age by 1', () => {
+    const pet = new Pet('Fido');
+    pet.growUp();
+    expect(pet.age).toEqual(1);
+    expect(pet.hunger).toEqual(5);
+  });
+});
+
+describe('walk', () => {
+  it('walk increases fitness by 4', () => {
+    const pet = new Pet('Fido');
+    pet.fitness = 4;
+    pet.walk();
+    expect(pet.fitness).toEqual(8);
+  });
+
+  it('walk increases fitness by 4 up to max of 10', () => {
+    const pet = new Pet('Fido');
+    pet.fitness = 8;
+    pet.walk();
+    expect(pet.fitness).toEqual(10);	
+  });
+});
+
+describe('feed', () => {
+
+  it('throws an exception if pet not alive', () => {
+    const pet = new Pet('Fido');
+    pet.age = 30;
+    expect(() => pet.feed()).toThrow('Your pet is no longer alive! :o(');
+  });
+
+  it('feed decreases hunger by 3', () => {
+    const pet = new Pet('Fido');
+    pet.hunger = 8;
+    pet.feed();
+    expect(pet.hunger).toEqual(5);
+  });	
+
+  it('feed decreases hunger by 3 down to minimum of 0', () => {
+  const pet = new Pet('Fido');
+    pet.hunger = 2;
+    pet.feed();
+    expect(pet.hunger).toEqual(0);
+    });	
+});
+
+describe('checkUp', () => {
+  it('return no longer alive', () => {
+    const pet = new Pet('Fido');
+    pet.age = 30;
+    expect(pet.checkUp()).toEqual('Your pet is no longer alive! :o(')		
```

Comment:
With multiline tests like this one, its convention to leave a blank line before the `expect` assertions to help differentiate them a bit from the setup code that comes before it